posts/2024-08-27-quarto-on-HPC/index.qmd

```diff
@@ -0,0 +1,166 @@
+
+Download the Quarto source code for *Linux x86 Tarball* from their [official website](https://quarto.org/docs/download/tarball.html?version=1.5.56&idPrefix=download) into your home directory.
+You can download the source code by copying the address link of the download, and using `wget` to download it directly into your home directory.
+
+::: {.callout-tip}
+Download the source code with `wget` using the following command (change `X.X.X` to the latest version available).
+```
+wget https://github.com/quarto-dev/quarto-cli/releases/download/vX.X.X/quarto-X.X.X-linux-amd64.tar.gz
+```
+:::
+
+::: {.callout-tip}
+If you are installing it in your PC, you can use the installer that matches your OS.
+:::
+
+Decompress the source code in a location of your preference within your home directory, like `/home/$USER/opt/`.
+
+::: {.callout-tip}
+You might need to create a `/home/$USER/opt` directory if it doesn't already exist.
+
+```
+mkdir /home/$USER/opt
+```
+:::
+
+::: {.callout-tip}
+Decompress the source code with `tar` by executing the following command (change the version `X.X.X` to what you are using first).
+
+```
+tar -C ~/opt -xvzf quarto-X.X.X-linux-amd64.tar.gz
+```
+:::
+
+Create a symbolic link to the Quarto executable inside your `/home/$USER/bin` directory.
+
+::: {.callout-tip}
+You might need to create a `/home/$USER/bin` directory if there is not one already as follows.
+
+```
+mkdir /home/$USER/bin
+```
+
+In general, your `/home/$USER/bin` directory should already be in your `PATH` environment variable, even if the directory didn't exist before.
+:::
+
+::: {.callout-tip}
+Create a symbolic link to the `quarto` executable with the following command (again, change the version `X.X.X` first).
+
+```
+ln -s /home/$USER/opt/quarto-X.X.X/bin/quarto /home/$USER/bin/quarto
+```
+:::
+
+Check that `quarto` is installed by executing the `quarto --version` command.
+
+
+## Option 2. Install Quarto in a `conda` virtual environment
+
+For this option, you should have `conda` installed in your home directory.
+I recommend using [miniforge](https://github.com/conda-forge/miniforge).
+
+Activate the virtual environment from where you will use Quarto, i.e. `conda activate my-venv` (change `my-venv` with the actual name of your environment).
+
+::: {.callout-note}
+If you don't have a virtual environment, start by creating one instead of installing packages in the `base` environment.
+:::
+
+Install `quarto` from the `conda-forge` channel as follows.
+
+```
+conda install -c conda-forge quarto
+```
+
+# Render and preview documents
+You can render and preview your Quarto markdowns by using the following command.
+
+```
+quarto preview --no-browser --no-watch-inputs
+```
+
+That will generate the rendered versions of your markdowns.
+For `html`, `revealjs`, or `websites` formats, it will start a local server and provide a link that you can paste in your browser to visualize the result.
+
+# Authoring Jupyter notebooks with Quarto

```

Comment:
This applies to other forms with code cells, like presentations too!
Maybe this heading should be made more generic, like `Authoring documents with code cells`

Reply:
I went for `Authoring documents with (Python) code cells` since I haven't tested with R, or any other language